### PR TITLE
Server for RPi

### PR DESCRIPTION
**Pull request by andrewyazura**

### CODE REVIEW

**Review by mishuill — APPROVED** [review]
OK.

**Comment by danbst** [comment]
Crash when running `curl -v 'localhost:8080/api/v1/debug/log?after=2020-01-01'`
```
2020-02-11 15:01:35,247 INFO: 127.0.0.1 - - [11/Feb/2020 15:01:35] "GET /api/v1/debug/log HTTP/1.1" 400 -
2020-02-11 15:02:07,046 ERROR: Exception on /api/v1/debug/log [GET]
Traceback (most recent call last):
  File "/nix/store/86aswhmhbmw92r4jrnk0h7j58z5rna4p-python3-3.7.6-env/lib/python3.7/site-packages/flask/app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "/nix/store/86aswhmhbmw92r4jrnk0h7j58z5rna4p-python3-3.7.6-env/lib/python3.7/site-packages/flask/app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/nix/store/86aswhmhbmw92r4jrnk0h7j58z5rna4p-python3-3.7.6-env/lib/python3.7/site-packages/flask/app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/nix/store/86aswhmhbmw92r4jrnk0h7j58z5rna4p-python3-3.7.6-env/lib/python3.7/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/nix/store/86aswhmhbmw92r4jrnk0h7j58z5rna4p-python3-3.7.6-env/lib/python3.7/site-packages/flask/app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "/nix/store/86aswhmhbmw92r4jrnk0h7j58z5rna4p-python3-3.7.6-env/lib/python3.7/site-packages/flask/app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "src/server/app.py", line 82, in get_logs
    timestamp = datetime.strptime(timestamp, "%Y-%m-%d %H:%M:%S,%f")
  File "/nix/store/s7vw8y02cqzx8bnjxl4bkhlnwvz6ws1s-python3-3.7.6/lib/python3.7/_strptime.py", line 577, in _strptime_datetime
    tt, fraction, gmtoff_fraction = _strptime(data_string, format)
  File "/nix/store/s7vw8y02cqzx8bnjxl4bkhlnwvz6ws1s-python3-3.7.6/lib/python3.7/_strptime.py", line 359, in _strptime
    (data_string, format))
ValueError: time data '2020-01-01' does not match format '%Y-%m-%d %H:%M:%S,%f'
2
```
Also please add request parameters to log file

**Comment by danbst** [comment]
@andrew4ever ping

File: src/server/services/move_servo.py [hunk 1]
```diff
@@ -1,5 +1,33 @@
...
+    angle: int
+        Angle servo will move to.
+    duration: int
+        Duration (in ms) servo will stay turned for.
+    """
     logging.info('Set servo to %i degrees for %ims', angle, duration)
+
+    move = multiprocessing.Process(
+        target=_move_servo, args=(angle, duration))
+    move.start()
```
Comment (danbst) [inline src/server/services/move_servo.py]: what will happen in this situation:
- move request with duration=1s comes up
- 0.5s delay
- another move request with duration=1s comes up

Will motors stop spinning after 0.5s or after 1s?

File: src/server/services/move_servo.py [hunk 2]
```diff
@@ -1,5 +1,33 @@
...
+    """Actually move servo to specified angle for some duration.
+
+    Parameters
+    ----------
+    angle: int
+        Angle servo will move to.
+    duration: int
+        Duration (in ms) servo will stay turned for.
+    """
+    time.sleep(duration)
```
Comment (danbst) [inline src/server/services/move_servo.py]: I guess there should be "stop motors" action after sleep

File: src/server/services/move_servo.py [hunk 3]
```diff
@@ -1,5 +1,33 @@
 import logging
+import multiprocessing
```
Comment (danbst) [inline src/server/services/move_servo.py:2]: please use threads, not processes
Or maybe even `concurrent.futures` module https://docs.python.org/3/library/concurrent.futures.html

File: src/server/ProcessManager.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+        job = self._jobs.get(tag, None)
+
+        if not job:
+            raise Exception('No such job')
+
+        if job['process'] and job['process'].is_alive():
+            job['process'].terminate()
+
+        job['process'] = multiprocessing.Process(
```
Comment (danbst) [inline src/server/ProcessManager.py:55]: should be thread, not process

Comment (andrewyazura) [reply src/server/ProcessManager.py:55]: Python threads or concurrent.futures that are already running can't be canceled or terminated

**Pull request merged**